platforms/database.go

```diff
@@ -0,0 +1,93 @@
+package platforms
+
+import (
+	"runtime"
+	"strings"
+)
+
+// validComboString defines valid GOOS/GOARCH combinations.
+// Should be updated from https://golang.org/doc/install/source#environment.
+// This is why we leave it as a string.
+const (
+	validComboString = `
```

Comment:
I think we only need `linux`, `windows`, and `solaris` entries here.
At least, `android`, `plan9`, `nacl`, and `zos` are really unlikely to be supported in containerd.

Also, I think it is ok to use typed struct here, as the upstream godoc list won't be updated so frequently.